emu_sv/hamiltonian.py

```diff
@@ -132,18 +133,19 @@ def _create_diagonal(self) -> torch.Tensor:
             diagonal elements.
         """
         diag = torch.zeros(
-            (2,) * self.nqubits, dtype=torch.complex128, device=self.omegas.device
+            2**self.nqubits, dtype=torch.complex128, device=self.deltas.device
         )
 
         for i in range(self.nqubits):
-            i_fixed = diag.select(i, 1)
-            i_fixed -= self.deltas[i]  # add the delta term for this qubit
+            diag = diag.reshape((2**i, 2, -1))  # -1 means "remaining"
```

Comment:
reshape takes a tuple but you can skip the extra parentheses also cuz it understands individual dims:
`diag = diag.reshape(2**i, 2, -1)`

also the comment is a bit superfluous, we use `-1` everywhere in the code

Reply:
done